src/main/scala/pipelines/text/NewsgroupsPipeline.scala

```diff
@@ -21,15 +21,17 @@ object NewsgroupsPipeline extends Logging {
 
     // Build the classifier estimator
     logInfo("Training classifier")
-    val predictor = Trim.then(LowerCase())
-        .then(Tokenizer())
-        .then(new NGramsFeaturizer[String](1 to conf.nGrams))
-        .then(TermFrequency(x => 1))
-        .thenEstimator(CommonSparseFeatures(conf.commonFeatures))
-        .fit(trainData.data)
-        .thenLabelEstimator(NaiveBayesEstimator(numClasses))
-        .fit(trainData.data, trainData.labels)
-        .then(MaxClassifier)
+    val predictorPipeline = Trim andThen LowerCase() andThen
+        Tokenizer() andThen
+        NGramsFeaturizer(1 to conf.nGrams) andThen
+        TermFrequency(x => 1) andThen
+        (CommonSparseFeatures(conf.commonFeatures), trainData.data) andThen
+        (NaiveBayesEstimator(numClasses), trainData.data, trainData.labels) andThen
+        MaxClassifier
+
+
+    val predictor = Optimizer.execute(predictorPipeline)
```

Comment:
Mmm... at the moment it will only make a difference in pipelines with > 1 estimator. We should add it to pipelines once we update them to construct the full DAGs before doing anything, but most of them currently don't.

As for when to call this automatically... still tbd, possibly when interpreting or compiling a pipeline, and @etrain may have some ideas.